backend/controllers/whereabouts/scheduledMoves.ts

```diff
@@ -22,14 +22,63 @@ export default ({ prisonApi, offenderSearchApi }) => {
         }))
         .sort(sortTextAlphabetically),
       courtEvents,
+      releaseEvents,
+      transferEvents,
     })
 
+  const getRelevantAlertFlagLabels = (alerts) => {
+    const relevantAlertCodesForScheduledMoves = alerts
+      .map((alert) => alert.alertCode)
+      .filter((alert) => relevantAlertsForTransfer.includes(alert))
+
+    return alertFlagLabels
+      .filter((flag) => flag.alertCodes.some((code) => relevantAlertCodesForScheduledMoves.includes(code)))
+      .map((alert) => ({
+        img: alert.img || null,
+        label: alert.label,
+        classes: alert.classes,
+      }))
+  }
+  const getScheduledMovementDetails = async (context, prisonerDetailsForOffenderNumbers) => {
+    const personalPropertyForPrisoners =
+      (await Promise.all(
+        prisonerDetailsForOffenderNumbers.flatMap((sr) =>
+          prisonApi.getPrisonerProperty(context, sr.bookingId).then((property) =>
+            property.map((prop) => ({
+              containerType: prop.containerType,
+              userDescription: formatPropertyDescription(prop.location.userDescription),
+              bookingId: sr.bookingId,
+              prisonerNumber: sr.prisonerNumber,
+            }))
+          )
+        )
+      )) || []
+
+    return prisonerDetailsForOffenderNumbers.map((details) => {
+      const personalProperty = personalPropertyForPrisoners
+        .flatMap((p) => p)
+        .filter((p) => p.prisonerNumber === details.prisonerNumber)
+        .map((p) => ({
+          containerType: p.containerType,
+          boxNumber: p.userDescription,
+        }))
+
+      return {
```

Comment:
Just a note to not forget that we have typescript available now - we have the option to use types/classes. It might make some of this stuff a bit easier to read/maintain.